core/io/resource_uid.cpp

```diff
@@ -242,6 +242,16 @@ void ResourceUID::clear() {
 	unique_ids.clear();
 	changed = false;
 }
+
+uint32_t ResourceUID::generate_random_u32() const {
+	uint32_t num;
+	Error err = ((CryptoCore::RandomGenerator *)crypto)->get_random_bytes((uint8_t *)&num, sizeof(uint32_t));
```

Comment:
Will this ever change to a int64? I've seen other systems like https://en.wikipedia.org/wiki/Snowflake_ID use 64 bits.

Reply:
I have a feeling 4,294,967,296 nodes in a single PackedScene is not an amount that will be reached any time soon.

Reply:
I didn't know we're incrementing. We are using random bytes here.

Reply:
Oh, then, it may actually be an issue, but unsure.

Reply:
It looks like where it is called, it checks for duplicates and calls again until a unique is found, so collisions shouldn't be a problem (aside from the time spent checking for duplicates).

Reply:
Probably no, this is only for a single scene so no need for so much.